FIRO_TSEnsembles/src/main/java/hec/stats/PercentilesComputable.java

```diff
@@ -0,0 +1,60 @@
+package hec.stats;
+
+public class PercentilesComputable implements Computable {
+    Double _percentile; //percentile in decimal
+
+    /**
+     * Instantiates a percentile computable object
+     * @param percentile is expected to be in decimal
+     */
+
+    public PercentilesComputable(Double percentile) {
+        this._percentile = percentile;
+    }
+
+    @Override
+    public float compute(float[] values) {
+        if (this._percentile > 1.0) {
+            throw new ArithmeticException("Percentile must be less than equal to 1");
+        }
+        if (this._percentile <= 0) {
+            throw new ArithmeticException("Percentile must be greater than 0");
+        }
+        //sorts array descending
+        for(int i = 0; i < values.length; i++) {
+            for(int j = i+1; j < values.length; j++) {
+                if( values[i] < values[j]) {
+                    float tmp = values[i];
+                    values[i] = values[j];
+                    values[j] = tmp;
+                }
+            }
+        }
+        //creates an array of probabilities - index position / n
+        int indexPosition = 0;
```

Comment:
if you know the number of values, and the percentile  startindex = floor(percentile*countofvalues) endindex= startindex+1

you can remove two loops between lines 34 and 49